streamingest: c2c UX fixes

lidorcarmel opened the pull request:
CREATE TENANT FROM REPLICATION: currently has an output (the job ids of the producer and consumer). This PR removes the output.

ALTER TENANT PAUSE/RESUME: currently prints the job id, and this PR removes the output.

ALTER TENANT COMPLETE: currently prints the job id. This PR changes the output to be the cutover timestamp. The rational is that the user may not know the cutover time because LATEST or NOW() etc. was used.

Fixes: #94706

Epic: CRDB-18752

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/93755)
<!-- Reviewable:end -->

stevendanna commented on pkg/ccl/streamingccl/streamingest/alter_replication_job.go:
```diff
@@ -25,13 +26,14 @@ import (
-var alterReplicationJobHeader = colinfo.ResultColumns{
-	{Name: "replication_job_id", Typ: types.Int},
+var alterReplicationCutoverHeader = colinfo.ResultColumns{
+	{Name: "cutover_time", Typ: types.TimestampTZ},
```
I think we lose precision here if we had a logical component to the cutover timestamp.

lidorcarmel replied:
interesting, thanks. I did not think about that.. we're also losing nanos when cutting over to a time from SHOW. I might nag you tomorrow about this if I don't have a better way.

lidorcarmel replied:
I guess we have at least a couple of options:
1. print a not-so-pretty hlc such as this:

```
root@127.0.0.1:26257/defaultdb> SELECT cluster_logical_timestamp();

    cluster_logical_timestamp
----------------------------------
  1671493837007535000.0000000000
(1 row)
```
2. print a less accurate timestamp (current pr) without logical and without nanos:
```
root@127.0.0.1:26257/defaultdb> alter tenant dest5 complete replication to latest;

          cutover_time
---------------------------------
  2022-12-06 00:25:02.098763+00
(1 row)
```
similar to:
```
root@127.0.0.1:26257/defaultdb> SELECT hlc_to_timestamp(cluster_logical_timestamp());

        hlc_to_timestamp
---------------------------------
  2022-12-19 23:50:55.100638+00
(1 row)
```

I was thinking that we need something that users can understand and not hlc.

in which scenarios are we going to have issues because of accuracy?

stevendanna replied:
> in which scenarios are we going to have issues because of accuracy?

I think the main use of this time in a context where the accuracy may matter is if you are using it to do comparisons using AOST queries or crdb_internal.fingeprint or similar.  In that context, the loss of accuracy is likely OK provided that the timestamp printed is inside the GC window on both sides and if we are inaccurate in the correct direction.

lidorcarmel replied:
updated, PTAL.
pretty time for replication and retained times, hlc for cutover.

Commit:
streamingest: remove job ids from CREATE TENANT FROM REPLICATION

Currently we print the producer and ingestion job ids. Instead,
users should not see job ids unless something is really broken.

Epic: CRDB-18752

Release note: None

Commit:
streamingest: fix ALTER TENANT REPLICATION output

ALTER TENANT PAUSE/RESUME: currently prints the job id, and this PR removes
the output.

ALTER TENANT COMPLETE: currently prints the job id. This PR changes the output
to be the cutover timestamp. The rational is that the user may not know the
cutover time because LATEST or NOW() etc. was used.

Epic: CRDB-18752

Release note: None

Commit:
sql: add timezone to SHOW TENANT WITH REPLICATION

And simplify the time conversion to datum a bit.

Epic: CRDB-18752

Release note: None

Commit:
sql: adjust replication min/max time

Currently the retained time (min, or oldest we can cutover to) and the
replicated time (max replicated time) are converted from an accurate
HLC to a human readable timestamp, rounded to microseconds.

Rounding means we might get the min or max time outside the time window,
where the user cannot cutover to that timestamp.

Instead or rounding this PR truncates the max timestamp and prints the
ceil of the min timestamp, so that both of those timestamps are legal
cutover timestmaps.

Epic: CRDB-18752

Release note: None

stevendanna reviewed (approved):
This seems reasonable to me. Thanks for digging into it.

lidorcarmel commented:
TFTR!!
bors r+
